controllers/purchaseorder.js

```diff
@@ -0,0 +1,224 @@
+						
+						return autoCallback(null, quotesCount);
+					});
+				},
+				// get the purchase order prifix
+				getPurchaseOrderPrifix: ['recordCount', function(result, autoCallback) {
+					var purchaseorderCount = (_.isArray(result.recordCount)) ? result.recordCount[0].tot : result.recordCount.tot;
+					if (purchaseorderCount === 0) return autoCallback();
+					
+					app.models.global_settings
+					.findOne({settingName:'purchaseorder_num_prefix'})
+					.exec(function(err,globalSetting) {
+						if (err) return autoCallback(err);
+						  
+						return autoCallback(null,globalSetting);
+					});
+				}],
+				// get the records
+				getRecords: ['recordCount', function(result, autoCallback) {
+					// if record count is 0 then no need to execute the query to get the data
+					var purchaseorderCount = (_.isArray(result.recordCount)) ? result.recordCount[0].tot : result.recordCount.tot;
+					if (purchaseorderCount === 0) return autoCallback();
+
+					var query = " select `purchase_order`.`idpurchase_order` as `id`,`purchase_order`.*,";
+						query+= " `purchase_order_custom_fld`.*,";
+						query+= " `purchase_order_address`.*,";
+						query+= " `vendor`.`vendor_name`,";
+						query+= " concat(`contacts`.`firstname`,' ',`contacts`.`lastname`) as `contact_name`,";
+						query+= " `purchase_order_to_grp_rel`.`idgroup`,";
+						query+= " case when (`user`.`user_name` not like '')";
+						query+= " then";
+						query+= " `user`.`user_name`";
+						query+= " else";
+						query+= " `group`.`group_name` end";
+						query+= " as `assigned_to`";
+						query+= " from `purchase_order`";
+						query+= " inner join `purchase_order_address` on `purchase_order_address`.`idpurchase_order` = `purchase_order`.`idpurchase_order`";
+						query+= " inner join `purchase_order_custom_fld` on `purchase_order_custom_fld`.`idpurchase_order` = `purchase_order`.`idpurchase_order`";
+						query+= " left join `user` on `user`.`iduser` = `purchase_order`.`iduser`";
+						query+= " left join `purchase_order_to_grp_rel` on `purchase_order_to_grp_rel`.`idpurchase_order` = `purchase_order`.`idpurchase_order`";
+						query+= " left join `vendor` on `vendor`.`idvendor` = `purchase_order`.`idvendor`";
+						query+= " left join `group` on `group`.`idgroup` = `purchase_order_to_grp_rel`.`idgroup`";
+						query+= " left join `contacts` on `contacts`.`idcontacts` = `purchase_order`.`idcontacts`";
+						query+= " where `purchase_order`.`deleted` = 0";
+						query+= whereClause;
+						query+= " order by `purchase_order`.`idpurchase_order`";
+						query+= " limit "+page+" , "+limit;
+					
+					app.models.purchaseorder
+					.query(query, function(err, purchaseorder) {
+						if (err) return autoCallback(err);
+						   
+						return autoCallback(null, purchaseorder);
+					});
+				}]
+			},
+			function(err, results) {
+				if (err) return next(err);
+				var purchaseorders = {},
+					purchseorderPrefix = results.getPurchaseOrderPrifix;
+				
```

Comment:
`purchseorderPrefix` there is a typo and its used later as `purchaseorderPrefix` line:163

JS hint shows the errors as 
```
163 |                                if (purchaseorderPrefix && purchaseorderPrefix.settingData && k === 'po_number') {
                                             ^ 'purchaseorderPrefix' is not defined.
    163 |                                if (purchaseorderPrefix && purchaseorderPrefix.settingData && k === 'po_number') {
                                                                    ^ 'purchaseorderPrefix' is not defined.
    164 |                                    v= purchaseorderPrefix.settingData+''+v;
                                                ^ 'purchaseorderPrefix' is not defined.

```